dandi/metadata.py

```diff
@@ -574,14 +600,27 @@ def extract_digest(metadata):
 }
 
 
-def extract_field(field, metadata):
+def extract_field(field: str, metadata: dict) -> Any:
     if field in FIELD_EXTRACTORS:
         return FIELD_EXTRACTORS[field](metadata)
     else:
-        return metadata.get(field, ...)
+        return metadata.get(field)
 
 
-neurodata_typemap = {
+if TYPE_CHECKING:
+    if sys.version_info[:2] >= (3, 8):
```

Comment:
It should work as a simpler
```suggestion
    if sys.version_info >= (3, 8):
```
right?